dev/org.eclipse.codewind.ui/src/org/eclipse/codewind/ui/internal/actions/CodewindInstall.java

```diff
@@ -0,0 +1,213 @@
+	
+	public static boolean isCodewindInstalled() throws InvocationTargetException {
+						
+			try {
+				ProcessResult result = InstallUtil.statusCodewind();
+
+				String resultOutput = result.getOutput().replace("\n", "").replace("\r", "");		
+				String isCodewindInstalled = resultOutput;
+					
+				if (isCodewindInstalled.equals(CODEWIND_NOT_INSTALLED)) {
+					return true;
+				} else {
+					return false;
+				}
+				
+			} catch (IOException e) {
+				throw new InvocationTargetException(e, "An error occurred trying to determine Codewind status: " + e.getMessage()); //$NON-NLS-1$
+			} catch (TimeoutException e) {
+				throw new InvocationTargetException(e, "Codewind did not return status in the expected time: " + e.getMessage()); //$NON-NLS-1$
+			}
+		
+	}
+
+
+
+	public static void codewindInstallerDialog() {
+		Shell shell = new Shell();
+		MessageBox dialog =
+			    new MessageBox(shell, SWT.ICON_QUESTION | SWT.YES| SWT.NO);
+			dialog.setText(Messages.InstallCodewindDialogTitle);
+			dialog.setMessage(Messages.InstallCodewindDialogMessage);
+
+			int rc = dialog.open();
+
+		    switch (rc) {
+		    case SWT.YES:
+				installCodewind();
+			 break;
+		    }
+	}
+	
+	public static void installCodewind() { 
+
+		try {
+			Job job = new Job(Messages.InstallCodewindJobLabel) {
+			    @Override
+				protected IStatus run(IProgressMonitor monitor) {
+
+					try {
+						
+						ProcessResult result = InstallUtil.installCodewind(monitor);
+						
+						if (monitor.isCanceled()) {
+							removeCodewind();
+							return Status.CANCEL_STATUS;
+						}
+						
+						if (result.getExitValue() != 0) {
+							return new Status(IStatus.ERROR, CodewindUIPlugin.PLUGIN_ID, "There was a problem trying to install Codewind: " + result.getError());
+						}
+						
+						if (result.getExitValue() == 0) {
+							startingCodewind();
+						}
+						
+					} catch (IOException e) {
+						return new Status(IStatus.ERROR, CodewindUIPlugin.PLUGIN_ID, "An error occurred trying to install Codewind.", e);
+					} catch (TimeoutException e) {
+						return new Status(IStatus.ERROR, CodewindUIPlugin.PLUGIN_ID, "Codewind did not install in the expected time.", e);
+					}
+
+					return Status.OK_STATUS;
+					
+				}
+			};
+			job.schedule();
+		} catch (Exception e) {
+			Logger.logError("An error occurred installing Codewind: ", e); //$NON-NLS-1$
+		}
+		
+	}
+	
+	public static void startingCodewind() { 
+
+		try {
+			Job job = new Job(Messages.StartingCodewindJobLabel) {
+			    @Override
+				protected IStatus run(IProgressMonitor monitor) {
+
+					try {
+						
+						ProcessResult result = InstallUtil.startCodewind(monitor);
+						
+						if (result.getExitValue() != 0) {
+							return new Status(IStatus.ERROR, CodewindUIPlugin.PLUGIN_ID, "There was a problem trying to start Codewind: " + result.getError());
+						}
+						
+						if (result.getExitValue() == 0) {
```

Comment:
The check for exit value 0 is not really needed here because of the above check with the return inside.